File: navigation/src/main/java/com/example/navigation/stepsEngine/flow/rules/comparable/string/StringContainsRule.java
package com.example.navigation.stepsEngine.flow.rules.comparable.string;

import com.example.navigation.stepsEngine.flow.rules.comparable.ComparableRule;
import com.example.navigation.stepsEngine.payment.FlowState;

public class StringContainsRule extends ComparableRule {

    @Override
    public boolean evaluate(FlowState flowState) {
        return ((String) getValue(flowState)).contains(getValue());
    }

}

File: navigation/src/main/java/com/example/navigation/stepsEngine/flow/rules/comparable/string/StringContainsRule.kt
package com.example.navigation.stepsEngine.flow.rules.comparable.string

import com.example.navigation.stepsEngine.flow.rules.comparable.ComparableRule
import com.example.navigation.stepsEngine.payment.FlowState

class StringContainsRule : ComparableRule() {
    override fun evaluate(flowState: FlowState): Boolean {
        return (getValue(flowState) as String?)!!.contains(value)
    }
}